File: xhrweb/EmployeeEnrollment.Core/Entities/EmployeeSalary.cs
using ASL.Hrms.SharedKernel;
using System;
using System.Collections.Generic;
using System.Text;

namespace EmployeeEnrollment.Core.Entities
{
    public class EmployeeSalary : BaseEntity<Guid>
    {
        public Guid? EmployeeId { get; private set; }
        public Guid? BranchId { get; private set; }
        public Guid? DepartmentId { get; private set; }
        public Guid? PositionId { get; private set; }
        public Guid? GradeId { get; private set; }
        public Guid? SalaryStructureId { get; private set; }
        public Int16? PaymentOptionId { get; private set; }
        public decimal? GrossSalary { get; private set; }
        public Guid? CompanyId { get; private set; }
        public DateTime? StartDate { get; private set; }
        public DateTime? EndDate { get; private set; }
        public Boolean IsDeleted { get; private set; }

        public EmployeeSalary(Guid id) : base(id) { }
        private EmployeeSalary() : base(Guid.NewGuid()) { }

    }
}

File: xhrweb/ASL.Hrms.Web/ClientApp/src/app/shared/models/payroll/employee-salary.ts
import { EmployeeSalaryComp } from "./employee-salary-comp";

export class EmployeeSalary {
  id: any;
  employeeId: any;
  branchId: any;
  departmentId: any;
  positionId: any;
  gradeId: any;
  salaryStructureId: any;
  paymentOptionId: any;
  grossSalary: any;
  companyId: any;
  startDate: any;
  endDate: any;
  branchName: any;
  departmentName: any;
  designationName: any;
  gradeName: any;
  structureName: any;
  optionName: any;
  fullName: any;
  branchIds: any;
  departmentIds: any;
  designationIds: any;
  positionName: string = "";
  companyEmployeeId: string = "";
  searchText: string = "";
  employeeSalaryComponentList: any[];
  public constructor(init?: Partial<EmployeeSalary>) {
    Object.assign(this, init);
  }
}
